cmd/mv.go

```diff
@@ -0,0 +1,116 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"reanahub/reana-client-go/client"
+	"reanahub/reana-client-go/client/operations"
+	"reanahub/reana-client-go/pkg/displayer"
+	"reanahub/reana-client-go/pkg/workflows"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const mvDesc = `
+Move files within workspace.
+
+The ` + "``close``" + ` command allows to move the files within workspace.
+
+Examples:
+
+	$ reana-client mv data/input.txt input/input.txt
+`
+
+type mvOptions struct {
+	token    string
+	workflow string
+	source   string
+	target   string
+}
+
+// newMvCmd creates a command to move files within workspace.
+func newMvCmd() *cobra.Command {
+	o := &mvOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "mv",
+		Short: "Move files within workspace.",
+		Long:  mvDesc,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			o.source = args[0]
+			o.target = args[1]
+			return o.run(cmd)
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&o.token, "access-token", "t", "", "Access token of the current user.")
+	f.StringVarP(
+		&o.workflow,
+		"workflow",
+		"w", "",
+		"Name or UUID of the workflow. Overrides value of REANA_WORKON environment variable.",
+	)
+
+	return cmd
+}
+
+func (o *mvOptions) run(cmd *cobra.Command) error {
+	statusPayload, err := workflows.GetStatus(o.token, o.workflow)
+	if err != nil {
+		return err
+	}
+	currentStatus := statusPayload.Status
+	if currentStatus == "running" {
+		return errors.New("file(s) could not be moved for running workflow")
+	}
+
+	files, err := workflows.ListFiles(o.token, o.workflow, "", "", 1, 0, false)
+	if err != nil {
+		return err
+	}
+	sourceExists := false
+	for _, file := range files.Items {
+		if strings.Contains(file.Name, o.source) {
+			sourceExists = true
+			break
+		}
+	}
+	if !sourceExists {
+		return fmt.Errorf("source file(s) %s does not exist in %s workspace", o.source, o.workflow)
+	}
```

Comment:
Should we keep this validation? I feel like we could just use the validation from the server but I wasn't sure so I ported the way it's done in Python

Reply:
Yes, I think it is a good idea to remove both of them, as they are already being done server side. However, the server side check that a workflow is not running is not working correctly, so I will open an issue about that.

Reply:
Related issue: https://github.com/reanahub/reana-workflow-controller/issues/469

Reply:
It's done, I had to update the openapi and regenerate the client